backtester/eventhandlers/statistics/currencystatistics/currencystatistics.go

```diff
@@ -40,26 +40,39 @@ func (c *CurrencyStatistic) CalculateResults() error {
 	}
 	c.MarketMovement = ((lastPrice - firstPrice) / firstPrice) * 100
 	c.StrategyMovement = ((last.Holdings.TotalValue - last.Holdings.InitialFunds) / last.Holdings.InitialFunds) * 100
+	c.calculateHighestCommittedFunds()
 	c.RiskFreeRate = last.Holdings.RiskFreeRate * 100
 	returnPerCandle := make([]float64, len(c.Events))
+	benchmarkRates := make([]float64, len(c.Events))
 
 	var allDataEvents []common.DataEventHandler
 	for i := range c.Events {
 		returnPerCandle[i] = c.Events[i].Holdings.ChangeInTotalValuePercent
+		if i == 0 {
+			benchmarkRates[i] = 0
+		} else {
+			benchmarkRates[i] = (c.Events[i].DataEvent.ClosePrice() - c.Events[i-1].DataEvent.ClosePrice()) / c.Events[i-1].DataEvent.ClosePrice()
```

Comment:
Might need to check for zero on `c.Events[i-1].DataEvent.ClosePrice()` when there is missing data? Maybe even return an error because this might skew everything else if I am reading this correctly.

Reply:
This one is a bit tough. I've left it up to the strategy implementation on how to handle signals when missing data, but modifying any data here is global. I think if a missing data event is encountered, I should put a big warning about how ratios are skewed and to get the data. Or all ratios = 0. What do you think?

Reply:
Yeah, it is hard well if this was 0 then we have a benchmark rate potentially being +inf. I don't the modifying of data. I think both options are good, I am leaning more towards ratios being 0.